consensus/src/simplex/actors/batcher/actor.rs

```diff
@@ -196,12 +196,15 @@ impl<
                             let is_active =
                                 // Not enough views have passed to judge activity
                                 current < View::new(self.skip_timeout.get())
-                                // We don't have enough history to judge activity
-                                || (work.len() as u64) < self.skip_timeout.get()
-                                // Leader was active in at least one recent round
-                                || work
-                                    .range(current.saturating_sub(self.skip_timeout)..)
-                                    .any(|(_, round)| round.is_active(leader));
+                                // Check leader activity in recent window
+                                || {
+                                    // Not enough recent history to judge (assume active)
+                                    let min_view = current.saturating_sub(self.skip_timeout);
+                                    let recent: Vec<_> = work.range(min_view..).collect();
+                                    (recent.len() as u64) < self.skip_timeout.get()
+                                    // Leader was active in at least one recent round
+                                    || recent.iter().any(|(_, round)| round.is_active(leader))
```

Comment:
Probably fine if `skip_timeout` is small, which I guess it should be, but I think ideally you just iterate once without collecting, and just increment a counter as you go, making sure that the counter at the end is equal-to-or-above skip-timeout. However, since this is such a strict requirement, this is also just equal to checking that the key is exactly equal to the previous key plus one (i.e. that there are no gaps). This strikes me as too strict.